### PR TITLE
Make additional text mime types optional

### PR DESCRIPTION
**Pull request by OverShifted**
Fixes a little "bug" in my clipboard manager which made it offer plain text mimes when copying images (because of `text/html`) which in turn caused some weirdness in some programs.

### CODE REVIEW

**Comment by YaLTeR** [comment]
> which made it offer plain text mimes when copying images (because of `text/html`)

How does this work, why do images have `text/html` type?

**Comment by OverShifted** [comment]
Sorry, I should've mentioned that this issue happens specifically when copying from Firefox. For some reason, Firefox copies the `img` tag's HTML code as `text/html`. When my clipboard manager tries to restore it, other mime types such as `text/plain` will be generated by `wl-clipboard-rs` which causes some programs to paste the HTML code (either instead, or alongside the actual image).

File: src/copy.rs [hunk 1]
```diff
@@ -142,6 +142,12 @@ pub struct Options {
     /// that certain apps may have issues pasting when this option is used, in particular XWayland
     /// clients are known to suffer from this.
     serve_requests: ServeRequests,
+
+    /// Offer additional text mime types if at least one text mime type is provided.
+    ///
+    /// Additionaly offers `text/plain;charset=utf-8`, `text/plain`, `STRING`, `UTF8_STRING` and
+    /// `TEXT` mime types if at least one text mime type is provided.
+    offer_additional_text_mimes: bool
```
Comment (YaLTeR) [inline src/copy.rs:150]: This will default to `false`, however I'd like to preserve the current behavior. I guess you should convert the Default derive into a manual impl and set this field to true. Alternatively, reword this to invert the meaning -- something like `omit_additional_text_mime_types`.

File: src/copy.rs [hunk 2]
```diff
@@ -284,6 +290,16 @@ impl Options {
         self
     }
 
+    /// Sets the flag for offering additional text mime types if at least one text mime type is provided.
+    ///
+    /// Additionaly offers `text/plain;charset=utf-8`, `text/plain`, `STRING`, `UTF8_STRING` and
```
Comment (YaLTeR) [inline src/copy.rs:295]:
```suggestion
    /// If set, additionally offers `text/plain;charset=utf-8`, `text/plain`, `STRING`, `UTF8_STRING` and
```

File: src/copy.rs [hunk 3]
```diff
@@ -142,6 +142,12 @@ pub struct Options {
     /// that certain apps may have issues pasting when this option is used, in particular XWayland
     /// clients are known to suffer from this.
     serve_requests: ServeRequests,
+
+    /// Offer additional text mime types if at least one text mime type is provided.
+    ///
+    /// Additionaly offers `text/plain;charset=utf-8`, `text/plain`, `STRING`, `UTF8_STRING` and
```
Comment (YaLTeR) [inline src/copy.rs:148]:
```suggestion
    /// Additionally offers `text/plain;charset=utf-8`, `text/plain`, `STRING`, `UTF8_STRING` and
```

File: src/copy.rs [hunk 4]
```diff
@@ -743,18 +759,20 @@ fn prepare_copy_internal(options: Options,
...
-                              "text/plain",
-                              "STRING",
-                              "UTF8_STRING",
-                              "TEXT"];
-            for &mime_type in &text_mimes {
-                // We don't want to overwrite an explicit mime type, because it might be bound to a
-                // different data_path
-                if !data_paths.contains_key(mime_type) {
-                    data_paths.insert(mime_type.to_string(), text_data_path.clone());
+        if options.offer_additional_text_mimes {
```
Comment (YaLTeR) [inline src/copy.rs:762]: Isn't this check redundant? `offer_additional_text_mimes` is already checked when setting `text_data_path` as far as I can see.

File: src/tests/copy.rs [hunk 5]
```diff
@@ -236,6 +236,7 @@ fn copy_multi_test() {
     let child = thread::spawn(move || {
         let mut opts = Options::new();
         opts.foreground(true);
+        opts.offer_additional_text_mimes(true);
```
Comment (YaLTeR) [inline src/tests/copy.rs:239]: Won't be needed once the current default behavior is preserved.

But I would also like a test for the new behavior.

**Review by YaLTeR — CHANGES_REQUESTED** [review]

File: src/copy.rs (hunk 1 above)
Comment (OverShifted) [reply src/copy.rs:150]: What about wrapping this specific field in a struct that defaults to true?

Comment (YaLTeR) [reply src/copy.rs:150]: Why? If we want to keep this as true = previous, then implementing Default manually will be the easiest.

**Review by YaLTeR — APPROVED** [review]

**Comment by OverShifted** [comment]
Why rustfmt is giving errors on files I haven't modified? While [it seems to be fine](https://github.com/YaLTeR/wl-clipboard-rs/actions/runs/3116709997/jobs/5054784903)?

**Comment by YaLTeR** [comment]
> Why rustfmt is giving errors on files I haven't modified when I run it locally, while [it seems to be fine](https://github.com/YaLTeR/wl-clipboard-rs/actions/runs/3116709997/jobs/5054784903)?

Are you using nightly? This project uses unstable rustfmt options.

**Comment by YaLTeR** [comment]
Thanks!
